### PR TITLE
resf [TASK][28536] Create UI

### PR DESCRIPTION
**Pull request by peartalent**
## Related Tickets
-https://edu-redmine.sun-asterisk.vn/issues/28536

## WHAT
- Create UI
## Evidence (Screenshot or Video)
- View home
image
- Item genres
image
- Item movie 
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/layout/fragment_home.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.recyclerview.widget.RecyclerView
+    android:background="@android:color/white"
+    android:id="@+id/recyclerHome"
+    xmlns:android="http://schemas.android.com/apk/res/android" android:layout_width="match_parent"
```
Comment (tuannm-1276) [inline app/src/main/res/layout/fragment_home.xml]: format code

File: app/src/main/res/layout/item_movie.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    <TextView
+        android:id="@+id/textDate"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignParentBottom="true"
+        android:layout_centerHorizontal="true"
+        android:layout_marginBottom="@dimen/dp_20"
+        android:lines="1"
+        android:text="Aug 22, 2020"
```
Comment (tuannm-1276) [inline app/src/main/res/layout/item_movie.xml]: hardcode

File: app/src/main/res/layout/item_movie.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    <TextView
+        android:id="@+id/textName"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignBottom="@+id/imageBackdrop"
+        android:layout_centerHorizontal="true"
+        android:layout_marginBottom="-70dp"
+        android:maxLines="1"
+        android:singleLine="true"
+        android:text="Spider man 3"
```
Comment (tuannm-1276) [inline app/src/main/res/layout/item_movie.xml]: hardcode

File: app/src/main/res/values/colors.xml [hunk 4]
```diff
@@ -6,4 +6,8 @@
     <color name="saffron">#F4C124</color>
     <color name="pomegranate">#F32424</color>
     <color name="amaranth">#E91E63</color>
+    <color name="gray_nurse">#EAEDEB</color>
```
Comment (tuannm-1276) [inline app/src/main/res/values/colors.xml]: tất cả các tên color đang đặt tên sai convention

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -8,15 +8,36 @@
     android:theme="@style/Theme.AppCompat"
     tools:context=".ui.main.MainActivity">
 
+    <androidx.appcompat.widget.Toolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/activity_main.xml]:
```suggestion
        android:layout_width="0dp"
```

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -8,15 +8,36 @@
...
+        android:layout_height="wrap_content"
+        android:background="@drawable/bg_toolbar"
+        android:theme="@style/Theme.AppCompat.DayNight"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.0"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/activity_main.xml]: attribute này đang bị thừa nha

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -8,15 +8,36 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:menu="@menu/menu_home"
+        app:title="@string/title_genres"
+        app:titleTextColor="@color/cape_cod" />
+
     <com.google.android.material.bottomnavigation.BottomNavigationView
         android:id="@+id/bottomNavigation"
         android:layout_width="match_parent"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/activity_main.xml]: chuyển thành 0dp và thêm constraint Start, End theo parent nha

File: app/src/main/res/layout/activity_main.xml [hunk 8]
```diff
@@ -8,15 +8,36 @@
...
+        app:itemBackground="@drawable/bg_bottom_navigation"
+        app:itemIconTint="@color/bnv_tab_item_foreground"
+        app:itemTextColor="@color/bnv_tab_item_foreground"
         app:layout_constraintBottom_toBottomOf="parent"
         app:menu="@menu/menu_bottom" />
+
+    <FrameLayout
+        android:id="@+id/frameView"
+        android:layout_width="@dimen/dp_409"
+        android:layout_height="@dimen/dp_617"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/activity_main.xml]: Không nên fix cứng size vì nhiều thiết bị dpi nhỏ sẽ hiển thị không đúng, sử dụng constraint và bias để xử lý trường hợp này nha

File: app/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -8,15 +8,36 @@
...
-        app:itemIconTint="@android:color/white"
-        app:itemTextColor="@android:color/white"
+        app:itemBackground="@drawable/bg_bottom_navigation"
+        app:itemIconTint="@color/bnv_tab_item_foreground"
+        app:itemTextColor="@color/bnv_tab_item_foreground"
         app:layout_constraintBottom_toBottomOf="parent"
         app:menu="@menu/menu_bottom" />
+
+    <FrameLayout
+        android:id="@+id/frameView"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/activity_main.xml]: tên không có nghĩa

File: app/src/main/res/layout/item_movie.xml [hunk 10]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/item_movie.xml]: mình suggest chuyển qua constraint layout nhé, relative cũ rồi và không linh hoạt

File: app/src/main/res/drawable/ic_search_black_24dp.xml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+<vector android:height="24dp" android:tint="#474047"
```
Comment (tuanpa-0908) [inline app/src/main/res/drawable/ic_search_black_24dp.xml]: format drawable

File: app/src/main/res/layout/fragment_home.xml [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+    android:layout_height="match_parent">
+
+</androidx.recyclerview.widget.RecyclerView>
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/fragment_home.xml]:
```suggestion
    android:layout_height="match_parent"/>
```

File: app/src/main/res/layout/item_movie.xml [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignBottom="@+id/imageBackdrop"
+        android:layout_marginStart="@dimen/dp_5"
+        android:layout_marginTop="@dimen/dp_5"
+        android:layout_marginEnd="@dimen/dp_5"
+        android:layout_marginBottom="-34dp"
+        android:background="@drawable/bg_circle"
+        android:padding="@dimen/dp_5"
+        android:text="9.0"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/item_movie.xml]: hard code

File: app/src/main/res/layout/item_genres.xml [hunk 14]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (huytq-2422) [inline app/src/main/res/layout/item_genres.xml]:
```suggestion
<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```

File: app/src/main/res/layout/item_movie.xml [hunk 15]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/relativeMovie"
```
Comment (huytq-2422) [inline app/src/main/res/layout/item_movie.xml]:
```suggestion
    android:id="@+id/constraintMovie"
```

File: app/src/main/res/values/dimens.xml [hunk 16]
```diff
@@ -1,10 +1,19 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
+    <dimen name="dp_0">0dp</dimen>
```
Comment (huytq-2422) [inline app/src/main/res/values/dimens.xml]: 0dp không phải hard code

**Review by aug16vcc — APPROVED** [review]

File: app/src/main/java/com/sunasterisk/movieintroduction/ui/main/MainActivity.kt [hunk 17]
```diff
@@ -3,16 +3,21 @@ package com.sunasterisk.movieintroduction.ui.main
...
 import com.google.android.material.bottomnavigation.BottomNavigationView
 import com.sunasterisk.movieintroduction.R
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/movieintroduction/ui/main/MainActivity.kt]: remove blank

File: app/src/main/res/layout/activity_main.xml [hunk 18]
```diff
@@ -8,15 +8,39 @@
...
+        app:itemIconTint="@color/bnv_tab_item_foreground"
+        app:itemTextColor="@color/bnv_tab_item_foreground"
         app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
         app:menu="@menu/menu_bottom" />
+
+    <FrameLayout
+        android:id="@+id/fragmentContainer"
+        android:layout_width="0dp"
```
Comment (thonc-2135) [inline app/src/main/res/layout/activity_main.xml]: frameCoainer chứ sao lại fragment nhỉ

File: app/src/main/res/layout/item_genres.xml [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+    android:layout_height="wrap_content"
+    android:layout_margin="@dimen/dp_10"
+    android:background="@android:color/transparent">
+
+    <Button
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/dp_150"
+        android:background="@drawable/bg_genres"
+        android:fontFamily="sans-serif-light"
+        android:text="Action"
```
Comment (thonc-2135) [inline app/src/main/res/layout/item_genres.xml]: hard code

File: app/src/main/res/layout/item_movie.xml [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:ignore="ContentDescription" />
+
+    <TextView
+        android:id="@+id/textVoteAverage"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignBottom="@+id/imageBackdrop"
+        android:layout_marginBottom="64dp"
```
Comment (thonc-2135) [inline app/src/main/res/layout/item_movie.xml]: hardcode

File: app/src/main/res/layout/item_movie.xml [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+        android:background="@drawable/bg_circle"
+        android:padding="@dimen/dp_5"
+        android:textColor="@color/colorSaffron"
+        android:textSize="@dimen/sp_20"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.047"
+        app:layout_constraintStart_toStartOf="parent"
+        tools:text="9.1" />
```
Comment (thonc-2135) [inline app/src/main/res/layout/item_movie.xml]: hard code

File: app/src/main/res/layout/item_movie.xml [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+
+    <TextView
+        android:id="@+id/textDate"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignParentBottom="true"
+        android:layout_centerHorizontal="true"
+        android:layout_marginBottom="@dimen/dp_20"
+        android:lines="1"
+        tools:text="Aug , 20, 2020"
```
Comment (thonc-2135) [inline app/src/main/res/layout/item_movie.xml]: hard code

File: app/src/main/res/layout/item_movie.xml [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+
+    <TextView
+        android:id="@+id/textName"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignBottom="@+id/imageBackdrop"
+        android:layout_centerHorizontal="true"
+        android:layout_marginTop="@dimen/dp_108"
+        android:maxLines="1"
+        tools:text="Spider man 1"
```
Comment (thonc-2135) [inline app/src/main/res/layout/item_movie.xml]: hard code

**Pull request merged**